xcc/connection.py

```diff
@@ -210,12 +210,15 @@ def ping(self) -> requests.Response:
         """
         return self.request(method="GET", path="/healthz")
 
-    def request(self, method: str, path: str, **kwargs) -> requests.Response:
+    def request(
+        self, method: str, path: str, *, headers: Optional[Dict[str, str]] = None, **kwargs
+    ) -> requests.Response:
```

Comment:
Why not leave the function signature intact and implement something like
```python
headers = kwargs.pop("headers", self.headers}
```
to handle the case where the caller specifies headers? It's a bit cleaner and more flexible and still permits the caller to "add" headers by writing code such as
```python
c = xcc.Connection.load()
r = c.request("GET", "/endpoint", headers={**c.headers, "X-Foo": "Bar"})
```